src/Psy/ParserFactory.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of Psy Shell
+ *
+ * (c) 2012-2014 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy;
+
+use PhpParser\Lexer;
+use PhpParser\Parser;
+use PhpParser\ParserFactory as OriginalParserFactory;
+use ReflectionClass;
+
+/**
+ * Parser factory to abstract over PHP parser library versions.
+ */
+class ParserFactory
+{
+    const ONLY_PHP5 = 'ONLY_PHP5';
+    const ONLY_PHP7 = 'ONLY_PHP7';
+    const PREFER_PHP5 = 'PREFER_PHP5';
+    const PREFER_PHP7 = 'PREFER_PHP7';
+
+    /**
+     * Available kinds for second version of PHP parser.
+     *
+     * @return array
+     */
+    private function getPossibleKinds()
+    {
+        $factory = new ReflectionClass(__CLASS__);
+
+        return array_values($factory->getConstants());
```

Comment:
I don't like to duplicate things this way (yet another string with the same constant names). But it's not so important for me, so if you think differently...